src/tests/testRefCount.cc

```diff
@@ -41,91 +42,123 @@ class AlsoRefCountable : public RefCountable, public _ToRefCount
 
     int doSomething() {
         if (!Instances)
-            exit(EXIT_FAILURE);
+            return 0;
         return 1;
     }
 };
 
-int
-main (int argc, char **argv)
+void
+testRefCount::testCountability()
 {
     {
         ToRefCount anObject(new _ToRefCount);
-        anObject->someMethod();
+        CPPUNIT_ASSERT_EQUAL(1, _ToRefCount::Instances);
+        CPPUNIT_ASSERT_EQUAL(1, anObject->someMethod());
         anObject = anObject;
+        CPPUNIT_ASSERT_EQUAL(1, _ToRefCount::Instances);
         ToRefCount objectTwo (anObject);
         anObject = objectTwo;
+        CPPUNIT_ASSERT_EQUAL(1, _ToRefCount::Instances);
         {
             ToRefCount anotherObject(new _ToRefCount);
             anObject = anotherObject;
+            CPPUNIT_ASSERT_EQUAL(2, _ToRefCount::Instances);
         }
 
         {
             ToRefCount aForthObject (anObject);
             anObject = ToRefCount(NULL);
-            aForthObject->someMethod();
+            CPPUNIT_ASSERT_EQUAL(2, _ToRefCount::Instances);
+            CPPUNIT_ASSERT_EQUAL(1, aForthObject->someMethod());
             aForthObject = NULL;
         }
+        CPPUNIT_ASSERT_EQUAL(1, _ToRefCount::Instances);
     }
+    CPPUNIT_ASSERT_EQUAL(0, _ToRefCount::Instances);
+}
 
+void
+testRefCount::testObjectToRefCounted()
+{
     /* Test creating an object, using it , and then making available as a
      * refcounted one:
      */
-    {
-        _ToRefCount *aPointer = new _ToRefCount;
-        aPointer->someMethod();
-        ToRefCount anObject(aPointer);
-    }
+    _ToRefCount *aPointer = new _ToRefCount;
+    CPPUNIT_ASSERT_EQUAL(1, aPointer->someMethod());
+    ToRefCount anObject(aPointer);
+    CPPUNIT_ASSERT_EQUAL(1, _ToRefCount::Instances);
+}
+
+void
+testRefCount::testStandalonePointer()
+{
     /* standalone pointers should be usable */
-    {
-        ToRefCount anObject;
-    }
+    ToRefCount anObject;
+    CPPUNIT_ASSERT_EQUAL(0, _ToRefCount::Instances);
+}
+
+void
+testRefCount::testCheckPointers()
+{
     /* Can we check pointers for equality */
-    {
-        ToRefCount anObject;
-        ToRefCount anotherObject(new _ToRefCount);
+    ToRefCount anObject;
+    ToRefCount anotherObject(new _ToRefCount);
 
-        if (anObject == anotherObject)
-            exit(1);
+    CPPUNIT_ASSERT(anObject != anotherObject);
 
-        anotherObject = NULL;
+    anotherObject = NULL;
```

Comment:
Please use nullptr instead of NULL on any touched lines.

And please also add a check on the Instances state being correct before this operation. Several of the other tests need the same change. To be sure the post-operation state is a valid test of the expected change the pre-operation state need to be confirmed.